src/services/session.service.ts

```diff
@@ -81,15 +80,27 @@ export class SessionService {
         );
       }
       
+      try {
+        await this.redisService.del(user.email);
+      } catch(e) {
+        throw new HttpError(
+          HttpStatusCode.UNAUTHORIZED,
+          'Invalid Token'
+        );
+      }
+      
       //update user as verified
-      user.isVerified = true;
-      this.redisService.removeVerificationToken(user.email);
-      this.userService.editUser(user.id, user);
+      try {
+        user.isVerified = true;
+        this.userService.editUser(user.id, user);

```

Comment:
I think editUser returns a promise right? if so, you could add await before this instruction so in case it fails, the error handling will be managed by the _Catch_ statement